### DISCUSSION TITLE
一些讨论

### DISCUSSION THREAD

**Discussion by northword** [zotero-chinese/wiki#93] [社区建设]
@zotero-cn/all 大家晚上好，

@l0o0 提出了一些想法，我想请大家一起讨论一下：

## 组织维护的内容
> 我之前的设想也和你们类似。我们这个组织包括了CSL和Zotero的资料，应该是包括下面这些
> + CSL中文手册
> + Zotero中文手册
> + Zotero 插件
> + Zotero 转换器

这一点之前在  https://github.com/orgs/zotero-cn/teams/developer/discussions/1 也有讨论过，我们目前只涉及了 CSL 资源和 Zotero 手册，我个人倾向于把转换器、样式、插件商店这些都逐步整合起来，但是我担心现有的人员维护不了（不过有更多开发者有意愿参加），而插件留在开发者下维护。不知道大家的想法？

## 邀请成员

https://github.com/orgs/zotero-cn/teams/all/discussions/1

一位 Zotero PDF Translate 等插件的开发者，和一位 B 站 UP 有参与的想法。

## 讨论方式

我们目前主要在组织的 [Discussions](https://github.com/orgs/zotero-cn/discussions)  里，和 [“全员团队”的讨论里](https://github.com/orgs/zotero-cn/teams/all) 进行讨论，我们有无必要建立 IM 通讯方式，比如 QQ 群？

## 域名

> 我们应该要有一个合适的域名，有了域名后，我们就有了网络上的根据地。相关资源要以挂靠在域名下，应该也方便我们后面扩展新的内容，同学们访问和登录也更方便，我们可能会更有一些主动权。

我们目前用的还是 GitHub Pages的域名：https://zotero-cn.github.io/ 。

我们有无注册域名的需求？

注册域名的话涉及备案和费用问题，可能主要是备案（如果是国内的域名交易商的话）。
还有一个问题在于，Gitee Pages是无法绑定自定义域名的，因此如果没有服务器，域名就只能绑定在GitHub上，无法解决GitHub、Gitee两个网址的问题。（也不知道GitHub部分同学无法访问能不能通过加jsdeliver或其他CDN解决）

## 目前的一些进展

- Styles 自动预览的脚本框架搭起来了，修修补补就应该可以用？
- csl 中文文档重新整理的进度大约完成了百分之个位数 orz
- Zotero 文档 markdown 版本估计得手动改一改

**Comment by zepinglee** [comment]
> 我们目前用的还是 GitHub Pages的域名：https://zotero-cn.github.io/ 。
> 我们有无注册域名的需求？
> 注册域名的话涉及备案和费用问题，可能主要是备案（如果是国内的域名交易商的话）。
还有一个问题在于，Gitee Pages是无法绑定自定义域名的，因此如果没有服务器，域名就只能绑定在GitHub上，无法解决GitHub、Gitee两个网址的问题。（也不知道GitHub部分同学无法访问能不能通过加jsdeliver或其他CDN解决）

可以用 gitee.io 吗？比如 <https://zotero-chinese.gitee.io/zotero-plugins/>，这样在目前没有捐款的情况下可以省下费用。

另外强烈建议把 `-cn` 后缀改成 `-zh`，尤其是在解决了一大堆 `language` 填 `cn` 的问题后，我对这个已经开始产生了强迫症。

> Styles 自动预览的脚本框架搭起来了，修修补补就应该可以用？

赞！

**Comment by northword** [comment]
> 可以用 gitee.io 吗？比如 https://zotero-chinese.gitee.io/zotero-plugins/ ，这样在目前没有捐款的情况下可以省下费用。
> 
gitee.io 也可以的，可以和 github 的同步用。不过 gitee 的内容审查比较严格，我之前镜像在 gitee 的笔记就不知道哪里的敏感词一直没放开 🌚
自己域名的费用的话，大约看根域名吧，有些还是蛮便宜的，一年可能十几二十块的，没有捐款倒也可以接受。

> 另外强烈建议把 -cn 后缀改成 -zh，尤其是在解决了一大堆 language 填 cn 的问题后，我对这个已经开始产生了强迫症。
> 
😂😂 可以讨论一下改成 zh 还是 chinese ，chinese 的话和现在 gitee 上的一致，zh 的话比较精简。

**Comment by ZMAlt** [comment]
> > 我之前的设想也和你们类似。我们这个组织包括了CSL和Zotero的资料，应该是包括下面这些
> >
> > CSL中文手册
> > Zotero中文手册
> > Zotero 插件
> > Zotero 转换器

> 这一点之前在 https://github.com/orgs/zotero-cn/teams/developer/discussions/1 也有讨论过，我们目前只涉及了 CSL 资源和 Zotero 手册，我个人倾向于把转换器、样式、插件商店这些都逐步整合起来，但是我担心现有的人员维护不了（不过有更多开发者有意愿参加），而插件留在开发者下维护。不知道大家的想法？

同意逐步维护样式，其他不熟悉。

> 我们有无注册域名的需求？
> 
> 注册域名的话涉及备案和费用问题，可能主要是备案（如果是国内的域名交易商的话）。
> 还有一个问题在于，Gitee Pages是无法绑定自定义域名的，因此如果没有服务器，域名就只能绑定在GitHub上，无法解决GitHub、Gitee两个网址的问题。（也不知道GitHub部分同学无法访问能不能通过加jsdeliver或其他CDN解决）

感觉没有很大必要弄域名；我倾向于只在 github 维护，放弃 gitee，减少工作量，而且这些东西现在受众其实比较小。

> 我们目前主要在组织的 Discussions 里，和 “全员团队”的讨论里 进行讨论，我们有无必要建立 IM 通讯方式，比如 QQ 群？

同意建立 QQ 群

> csl 中文文档重新整理的进度大约完成了百分之个位数 orz

这个我觉得可以慢慢来，现在基本东西都有。我后半年有空也一起弄。

> Zotero 文档 markdown 版本估计得手动改一改

这个我感觉现在稍微有点麻烦。韩老师那边改动以后，这边要怎么同步。这边同步以后还要重新转成 mkdown，然后编译部署，有些繁琐。

**Comment by zepinglee** [comment]
> 感觉没有很大必要弄域名；我倾向于只在 github 维护，放弃 gitee，减少工作量，而且这些东西现在受众其实比较小。

如果我没理解错的话，用户下载样式是需要从 GitHub 下载的。由于 GitHub 经常被墙，对于没梯子的用户就很不方便。
 
> 我们目前主要在组织的 Discussions 里，和 “全员团队”的讨论里 进行讨论，我们有无必要建立 IM 通讯方式，比如 QQ 群？
> 
> 同意建立 QQ 群

我其实更喜欢 issue/discussion 的方式。

> Zotero 文档 markdown 版本估计得手动改一改
> 
> 这个我感觉现在稍微有点麻烦。韩老师那边改动以后，这边要怎么同步。

目前用的 submodule，我希望后续将那个仓库 transfer 过来。

> 这边同步以后还要重新转成 mkdown，然后编译部署，有些繁琐。

后续可以用 Github Action，每次 commit 后自动编译部署。

**Comment by ZMAlt** [comment]
> 目前用的 submodule，我希望后续将那个仓库 transfer 过来。

对，换到这边维护才好一点。如果所有的静态站都只在 github 维护的话，可以用 Action 同时出 pdf 和 html，pdf  给没梯子的。

**Comment by northword** [comment]
忘了说一下了， @l0o0 您是可以直接给 GitHub 用户发送邀请链接的，[步骤在这里](https://docs.github.com/cn/organizations/managing-membership-in-your-organization/inviting-users-to-join-your-organization) 。

**Comment by northword** [comment]
> 如果我没理解错的话，用户下载样式是需要从 GitHub 下载的。由于 GitHub 经常被墙，对于没梯子的用户就很不方便。
> 对，换到这边维护才好一点。如果所有的静态站都只在 github 维护的话，可以用 Action 同时出 pdf 和 html，pdf 给没梯子的。

Gitee 那边只镜像就好，不接受 Gitee 端的 commit，这样就可以了。

> 这个我感觉现在稍微有点麻烦。韩老师那边改动以后，这边要怎么同步。这边同步以后还要重新转成 mkdown，然后编译部署，有些繁琐。
> 目前用的 submodule，我希望后续将那个仓库 transfer 过来。
> 这边同步以后还要重新转成 mkdown，然后编译部署，有些繁琐。
> 后续可以用 Github Action，每次 commit 后自动编译部署。

目前是韩老师仓库在这边是子模块，之需要更新一下模块， GitHub Action 会自动转。
问题在于，自动转的效果不是特别理想（对于手动改出来的，专门用了静态页面生成器的专有语法的而言），所以我想的是，可以做两份，一份是 Tex 转 出来的，完全不手动改，只随着韩老师更新而自动更新（或者存档了）。另一份可以基于韩老师的现有的，专门手动改一个 Markdown 的，内容不必与韩老师保持一致。

**Comment by l0o0** [comment]
> 如果我没理解错的话，用户下载样式是需要从 GitHub 下载的。由于 GitHub 经常被墙，对于没梯子的用户就很不方便。
>     对，换到这边维护才好一点。如果所有的静态站都只在 github 维护的话，可以用 Action 同时出 pdf 和 html，pdf 给没梯子的。
我也觉得这样好，我们就在github，gitee只是备份，作为国内镜像方便下载。不过我担心后面gitee可能限制越来越多。

我之前想申请一个域名，作为我们对外的窗口，方便我们国内的同学找到我们。像Gitee Page的网页，我觉得目前还是有一些限制，它的一些功能我担心以后会收费或有其他限制。如果我们换成其他gitlab的话，那像https://zotero-chinese.gitee.io/zotero-plugins/#/ 这样的地址又会改变了，使用上可能不够友好。有了域名之后，我们可以做的事情就很多，不仅可以挂上各类资料，我们这边的主动性会更强一些。

**Comment by northword** [comment]
> 我之前想申请一个域名，作为我们对外的窗口，方便我们国内的同学找到我们。像Gitee Page的网页，我觉得目前还是有一些限制，它的一些功能我担心以后会收费或有其他限制。如果我们换成其他gitlab的话，那像https://zotero-chinese.gitee.io/zotero-plugins/#/ 这样的地址又会改变了，使用上可能不够友好。有了域名之后，我们可以做的事情就很多，不仅可以挂上各类资料，我们这边的主动性会更强一些。

我也倾向有一个域名，但是域名指向哪里 —— CNAME 到 Gitee 要收费，CNAME 到 GitHub 部分人登不上，A 到云服务器的花费就大了，划不来。301/302 的话并不能完全解决地址改变问题 —— 其他人收藏的链接大概率会是跳转后的，而不是注册的域名，能意识到应该要保存跳转前域名的大概率也可以解决 GitHub 问题。

@ZMAlt 请教一下 GMT 中文社区 是怎么解决这个问题的呢？似乎是仅指向了 GitHub Pages？

**Comment by ZMAlt** [comment]
是的，是 github。GMT 中文社区是一个非常小的圈子，基本都是地学领域的。目前基本上默认地学搞科研的人梯子是必备，访问不是问题。手册生成相应的 pdf 和 html。我们有两个 2000 人的 QQ 群，pdf 和维护的其他数据都会放到群里。
zotero 等的圈子还是要大一些，尤其是社会科学领域的，估计很难科学上网。
